Libraries/LibC/stdio.cpp

```diff
@@ -911,20 +910,26 @@ void perror(const char* s)
 static int parse_mode(const char* mode)
 {
     int flags = 0;
+
+    if (*mode == 'e') {
+        flags = O_CLOEXEC;
+        ++mode;
```

Comment:
As with every other gnu extension, there's no particular definition of where this goes or how it goes where it goes, so I just assumed it goes at the beginning, since that was the most clean way :shrug:

Reply:
I've personally written code that calls both `::fopen(filename, "re" )` and `::fopen(filename, "we")` and they both seem to work? Not sure that's a good assumption ;)

Reply:
Well that's concerning...
how do these work anyway? does, for instance, "re+" mean the same thing as "r+e"?
"ab" ~~ "ba"?

If so, it's just a slightly weird thing that can be done relatively easily...if not, I don't like implementing it.

Reply:
In the bugs section of this page
https://www.man7.org/linux/man-pages/man3/fopen.3.html#bugs

It says that it will parse "up to 7 characters". So I think you're allowed to put them in whatever order you want.

Reply:
If this random repository is to be believed, the first character has to be one of rwa, the rest can be in any order
https://github.com/lattera/glibc/blob/895ef79e04a953cac1493863bcae29ad85657ee1/libio/fileops.c#L225

Reply:
Let's hope this change doesn't break anything :P

Reply:
> As with every other gnu extension, there's no particular definition of where this goes or how it goes where it goes, so I just assumed it goes at the beginning, since that was the most clean way shrug

The man page says:

> The argument mode points to a string beginning with one of the following sequences (possibly followed by additional characters, as described below).

> The mode string can also include the letter 'b' either as a last character or as a character between the characters in any of the two-character strings described above.

And lists "r", "r+", "w", w+", "a", "a+" as those sequences it has to begin with, and "c", "e", "m", "x" (and ",ccs=...") as additional characters.

So I don't think your criticism of GNU extensions is justified, at least not in this case.

Reply:
Now that I think of it, and read your patch, — why don't we just support any letters in any order? Just loop over them and `|=` the required flags.